packages/react-components/src/components/utils/filtrer-tree/table-row.tsx

```diff
@@ -32,13 +32,14 @@ export class TableRow extends React.Component<TableRowProps> {
     };
 
     renderToggleCollapse = (): React.ReactNode => {
-        const marginLeft = this.props.level * 15 + 'px';
+        const width = (this.props.level + 1) * 12 + 'px';
         return (
             (this.props.node.children.length === 0)
-                ? <span style={{ marginLeft: marginLeft, paddingLeft: 20 }}></span>
-                : <span style={{ marginLeft: marginLeft, paddingRight: 5 }} onClick={this.handleCollapse}>
-                    {(this.isCollapsed() ? icons.expand : icons.collapse)}
-                </span>
+                ? <div style={{ width: width, paddingRight: 5 }}>&nbsp;</div>
+                : <div style={{ width: width, paddingRight: 5, display: 'flex' }} onClick={this.handleCollapse}>
+                    <div style={{ marginLeft: 'auto', marginRight: 0 }}>{(this.isCollapsed() ? icons.expand : icons.collapse)}
+                    </div>
+                </div>
```

Comment:
Works with inline:
```
(this.props.node.children.length === 0
  ? <div style={{ marginLeft: indentation + 12, display: 'inline' }} />
  : <div style={{ marginLeft: indentation, paddingRight: 5, display: 'inline' }} onClick={this.handleCollapse}>
    <div style={{ marginLeft: 'auto', marginRight: 0, display: 'inline' }}>
      {(this.isCollapsed() ? icons.expand : icons.collapse)}
    </div>
  </div>
)
```

Reply:
This reverts the fix because it puts back the variable icon width into the indentation. But it's on the right track, using 'inline-block' the desired indentation width can be respected. We can also set text-align: right to align the icon inside.